src/Unleash/Internal/UnleashServices.cs

```diff
@@ -20,34 +18,43 @@ internal class UnleashServices : IDisposable
 
         internal CancellationToken CancellationToken { get; }
         internal IUnleashContextProvider ContextProvider { get; }
-        internal ThreadSafeToggleCollection ToggleCollection { get; }
         internal bool IsMetricsDisabled { get; }
-        internal ThreadSafeMetricsBucket MetricsBucket { get; }
         internal FetchFeatureTogglesTask FetchFeatureTogglesTask { get; }
-
-        public UnleashServices(UnleashSettings settings, EventCallbackConfig eventConfig, Dictionary<string, IStrategy> strategyMap)
+        internal YggdrasilEngine engine { get; }
+        private static readonly List<string> DefaultStrategyNames = new List<string> {
+            "applicationHostname",
+            "default",
+            "flexibleRollout",
+            "gradualRolloutRandom",
+            "gradualRolloutSessionId",
+            "gradualRolloutUserId",
+            "remoteAddress",
+            "userWithId"
+        };
+
+        public UnleashServices(UnleashSettings settings, EventCallbackConfig eventConfig, List<IStrategy> strategies = null)
         {
             if (settings.FileSystem == null)
             {
                 settings.FileSystem = new FileSystem(settings.Encoding);
             }
 
+            engine = new YggdrasilEngine(strategies);
+
             var backupFile = settings.GetFeatureToggleFilePath();
             var etagBackupFile = settings.GetFeatureToggleETagFilePath();
 
             // Cancellation
             CancellationToken = cancellationTokenSource.Token;
             ContextProvider = settings.UnleashContextProvider;
 
-            var loader = new CachedFilesLoader(settings.JsonSerializer, settings.FileSystem, settings.ToggleBootstrapProvider, eventConfig, backupFile, etagBackupFile, settings.BootstrapOverride);
+            var loader = new CachedFilesLoader(settings.FileSystem, settings.ToggleBootstrapProvider, eventConfig, backupFile, etagBackupFile, settings.BootstrapOverride);
             var cachedFilesResult = loader.EnsureExistsAndLoad();
 
-            ToggleCollection = new ThreadSafeToggleCollection
+            if (!string.IsNullOrEmpty(cachedFilesResult.InitialState))
             {
-                Instance = cachedFilesResult.InitialToggleCollection ?? new ToggleCollection()
-            };
-
-            MetricsBucket = new ThreadSafeMetricsBucket();
+                engine.TakeState(cachedFilesResult.InitialState);
```

Comment:
What should we do if this fails? This can fail if the bootstrap are borked but I don't think this should bubble the exception all the way up - open to other thoughts here

Reply:
We discussed that the best idea we have for now is to wrap this in a try catch and log it. Addressed in https://github.com/Unleash/unleash-client-dotnet/pull/224/commits/4ff6c892d8de8824f21ae333fcc6585418a1fbe0